src/SFA.DAS.Authorization/AuthorizationRegistry.cs

```diff
@@ -13,6 +14,9 @@ public AuthorizationRegistry()
             For<IAuthorizationService>().Use<AuthorizationService>();
             For<ILoggerFactoryManager>().Use(c => new LoggerFactoryManager(c.TryGetInstance<ILoggerFactory>())).Singleton();
             For(typeof(ILogger<>)).Use(typeof(Logger<>)).Ctor<ILoggerFactory>().Is(c => c.GetInstance<ILoggerFactoryManager>().GetLoggerFactory());
+
+            // For web apps also need services.AddMemoryCache(); in start up
```

Comment:
Could `AddMemoryCache()` be called in `ServiceCollectionExtensions`? It looks like it uses `TryAdd` under the hood so if it's called multiple times (e.g. by the consuming application and the authorization package) it shouldn't cause any issues. Alternatively could the `MVC Core` section of the readme be updated to include a note that consuming applications will need to add this line to their application startup.

Also `IServiceCollection` typically won't be available in .NET Framework applications of which this package is also currently consumed. Could we register the caching components here explicitly as a workaround? e.g.

```c#
#if NET462
For(typeof(IOptions<>)).Use(typeof(OptionsManager<>)).Singleton();
For(typeof(IOptionsSnapshot<>)).Use(typeof(OptionsManager<>));
For(typeof(IOptionsMonitor<>)).Use(typeof(OptionsMonitor<>)).Singleton();
For(typeof(IOptionsFactory<>)).Use(typeof(OptionsFactory<>));
For(typeof(IOptionsMonitorCache<>)).Use(typeof(OptionsCache<>)).Singleton();
For<IMemoryCache>().Use<MemoryCache>().Singleton();
#endif
```

Alternatively we could do one of:

* Add a .NET Framework specific concrete implementation of `IAuthorizationCacheService`.
* Set the `SFA.DAS.Authorization.CommitmentPermissions` project/package to be .NET Core instead of .NET Standard.
* Set all `SFA.DAS.Authorization` projects/packages to be .NET Core instead of .NET Standard, effectively retiring .NET Framework support.